Sometimes fraction input doesn't let you submit

Just caught this for one scenario: I tried entering '2' and it failed:

```
2019-11-19 23:50:13.886 20588-20588/org.oppia.app E/StateFragment: Failed to retrieve answer outcome
    java.lang.IllegalStateException: Failed when classifying answer # org.oppia.app.model.InteractionObject@7485258e
    fraction {
      denominator: 1
      whole_number: 2
    } for interaction FractionInput
        at org.oppia.domain.classify.AnswerClassificationController.classifyAnswer(AnswerClassificationController.kt:51)
        at org.oppia.domain.classify.AnswerClassificationController.classify$domain_debug(AnswerClassificationController.kt:29)
        at org.oppia.domain.exploration.ExplorationProgressController.submitAnswer(ExplorationProgressController.kt:130)
        at org.oppia.app.player.state.StateFragmentPresenter.handleSubmitAnswer(StateFragmentPresenter.kt:484)
        at org.oppia.app.player.state.StateFragmentPresenter.onSubmitButtonClicked(StateFragmentPresenter.kt:475)
        at org.oppia.app.player.state.itemviewmodel.StateNavigationButtonViewModel.triggerContinuationNavigationButtonCallback(StateNavigationButtonViewModel.kt:81)
        at org.oppia.app.databinding.StateButtonItemBindingImpl._internalCallbackOnClick(StateButtonItemBindingImpl.java:439)
        at org.oppia.app.generated.callback.OnClickListener.onClick(OnClickListener.java:11)
        at android.view.View.performClick(View.java:6597)
        at android.view.View.performClickInternal(View.java:6574)
        at android.view.View.access$3100(View.java:778)
        at android.view.View$PerformClick.run(View.java:25885)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.IllegalStateException: Expected input value to be of type NON_NEGATIVE_INT not REAL
        at org.oppia.domain.classify.rules.GenericRuleClassifier.retrieveInputObject(GenericRuleClassifier.kt:36)
        at org.oppia.domain.classify.rules.GenericRuleClassifier.matches(GenericRuleClassifier.kt:25)
        at org.oppia.domain.classify.AnswerClassificationController.classifyAnswer(AnswerClassificationController.kt:46)
```

It didn't end up permanently broken, though, so I'm not sure this is the same problem. I'm also confused why the InteractionObject type is being mixed up...it should always be a fraction.

--- a/domain/src/main/java/org/oppia/domain/util/StateRetriever.kt
+++ b/domain/src/main/java/org/oppia/domain/util/StateRetriever.kt
@@ -184,7 +184,7 @@ class StateRetriever @Inject constructor(
           "HasDenominatorEqualTo", "HasNumeratorEqualTo" -> ruleSpecBuilder.putInput(
             inputName,
             InteractionObject.newBuilder()
-              .setReal(inputsJson.getDouble(inputName))
+              .setNonNegativeInt(inputsJson.getInt(inputName))
               .build()
           )
           else -> ruleSpecBuilder.putInput(inputName, createExactInputFromJson(inputsJson, inputName, interactionId))

--- a/domain/src/main/java/org/oppia/domain/util/StateRetriever.kt
+++ b/domain/src/main/java/org/oppia/domain/util/StateRetriever.kt
@@ -181,7 +181,13 @@ class StateRetriever @Inject constructor(
       while (inputKeysIterator.hasNext()) {
         val inputName = inputKeysIterator.next()
         when (ruleSpecBuilder.ruleType) {
-          "HasDenominatorEqualTo", "HasNumeratorEqualTo" -> ruleSpecBuilder.putInput(
+          "HasNumeratorEqualTo" -> ruleSpecBuilder.putInput(
+            inputName,
+            InteractionObject.newBuilder()
+              .setSignedInt(inputsJson.getInt(inputName))
+              .build()
+          )
+          "HasDenominatorEqualTo" -> ruleSpecBuilder.putInput(
             inputName,
             InteractionObject.newBuilder()
               .setNonNegativeInt(inputsJson.getInt(inputName))

